packages/legacy/src/queries/tree.ts

```diff
@@ -84,29 +85,39 @@ export const useInventoryTreeQuery = <T extends InventoryTree>(
   treeType: InventoryTreeType
 ): UseQueryResult<IndexedTree<T>> => {
   const indexTreeCallback = React.useCallback((data): IndexedTree<T> => indexTree(data), []);
-  // VMware providers have both Host and VM trees, but RHV only has Host trees.
-  const isValidQuery = provider?.type === 'vsphere' || treeType === InventoryTreeType.Cluster;
-  const apiSlug =
-    treeType === InventoryTreeType.Cluster
-      ? provider?.type === 'vsphere'
-        ? '/tree/host' // TODO in the future, this vsphere tree will also be at /tree/cluster
-        : '/tree/cluster'
-      : '/tree/vm';
+  // only VMware providers have both Host and VM trees.
+  const isInvalidQuery = treeType === InventoryTreeType.VM && provider?.type !== 'vsphere';
+  const apiSlugByProviderType: Record<ProviderType, string> = {
+    vsphere: '/tree/host',
+    ovirt: '/tree/cluster',
+    openstack: '/tree/project',
+    openshift: ''
+  }
+  const mockTreeData: Record<ProviderType, IInventoryHostTree> = {
+    vsphere: MOCK_VMWARE_HOST_TREE,
+    ovirt: MOCK_RHV_HOST_TREE,
+    openstack: MOCK_OPENSTACK_HOST_TREE,
+    openshift: undefined
+  }
+
+  const apiSlug = (treeType === InventoryTreeType.Cluster)
+    ? apiSlugByProviderType[provider?.type]
+    : '/tree/vm';
+    
+  const apiMockData = (treeType === InventoryTreeType.Cluster)
+    ? mockTreeData[provider?.type]
+    : MOCK_VMWARE_VM_TREE;
 
   return useMockableQuery<T, unknown, IndexedTree<T>>(
     {
       queryKey: ['inventory-tree', provider?.name, treeType],
       queryFn: async () =>
         await consoleFetchJSON(getInventoryApiUrl(`${provider?.selfLink || ''}${apiSlug}`)),
-      enabled: isValidQuery && !!provider,
+      enabled: !isInvalidQuery && !!provider,
```

Comment:
double negation here could be avoided - can we just name the flag `isVMwareClusterTree` ?